core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -541,6 +542,131 @@ class RemoteIndexCacheTest {
     verifyFetchIndexInvocation(count = 1)
   }
 
+  @Test
+  def testIndexFileAlreadyExistOnDiskButNotInCache(): Unit = {
```

Comment:
Yes @showuon  , this can happen in real case
1. Call getIndexEntry 
2. Offset index is successfully cached 
3. But TimeIndex received from remote storage is corrupted 
4. getIndexEntry throws Corrupt Exception
But it leaves the offset index file on disk in remote cache directory.

Reply:
Hmm... make sense! So, don't you think we should make the test case go through the above steps 1 to 4, and then verify next getIndexEntry can still work fine?

Reply:
Maybe we can merge it into the `testRSMReturnCorruptedIndexFile` test case?

Reply:
Yes @showuon  We can merge this in "testCorruptOffsetIndexFileExistsButNotInCache"  , instead of "testRSMReturnCorruptedIndexFile"

Reply:
@showuon  This use case is already covered in testIndexFileAlreadyExistOnDiskButNotInCache  , the only difference is files was created manually rather than the above flow ? Do you still see a need to merge this ?

Reply:
Yes, I think test the real flow makes more sense. We can change the `testIndexFileAlreadyExistOnDiskButNotInCache` into the flow above. WDYT?

Reply:
@showuon  I can write another test case for the same , as  testIndexFileAlreadyExistOnDiskButNotInCache is covering all index files existence .
The above scenario will be reproduce only in case of exception thrown  after offsetindex file is created i.e timeindex corruption or any other exception. WDYT ?

Reply:
@showuon  please review added the test case discussed above 
**testMultipleIndexEntriesExecutionInCorruptException**